syntax/hcl.vim

```diff
@@ -41,11 +41,13 @@ syn region hclComment start=/\/\*/ end=/\*\// contains=hclTodo
 syn match hclAttributeName /\w\+/ contained
 syn match hclAttribute     /^.*=/ contains=hclAttributeName,hclComment,hclString
 
-syn match hclBlockName /\w\+/ contained
-syn match hclBlock     /^[^=]\+{/ contains=hclBlockName,hclString
+syn match hclBlockName /\<[A-Za-z0-9_.\[\]*]\+\>/ nextgroup=hclString,hclBlock
+syn region hclBlock start="{" end="}" fold transparent contains=hclBlock,hclVariable,hclString,hclInterpolation,hclComment
+syn region hclBlock start="\[" end="\]" fold transparent contains=hclBlock,hclVariable,hclString,hclInterpolation,hclComment
```

Comment:
I think the change in `hclBlock` breaks the highlighting of `hclAttributeName`, which used to be unadorned (i.e. light gray) but after this change is highlighted the same way as `hclVariable` (i.e. yellow). This, I feel, makes the syntax highlighting worse, as a typical attribute name as well as a typical attribute value end up both being highlighted the same way.

Reply:
I think this change also breaks the highlighting of `hclNumber`, which is not included in `hclBlock`.